src/pin.ts

```diff
@@ -220,15 +192,59 @@ class ContractPinner {
   }
 
   /**
-   * Copy contract files to the version directory
-   * This includes both the main contract and its slim version if it exists
+   * Copy contract files from source to target directory
+   * Reads file names from manifest if available, falls back to contract path
    */
   private async copyContractFiles (contractPath: string, contractName: string, version: string) {
     const sourceDir = dirname(join(this.projectRoot, contractPath))
     const targetDir = join(this.projectRoot, 'contracts', contractName, version)
 
+    // Look for manifest file to get actual file names
+    // Scan directory for manifest files matching pattern: <contractName>.*.manifest.json
+    let manifestSource: string | null = null
+
+    try {
+      const { readdir } = await import('node:fs/promises')
+      const files = await readdir(sourceDir)
+      const manifestPattern = new RegExp(`^${contractName}\\..*\\.manifest\\.json$`)
```

Comment:
1. If you create a regular expression like this, the user-controlled inputs need to be properly escaped.
2. You likely don't need any of this logic here anyhow if you take the manifest file as an input for the command directly. That's what makes most sense in this case IMHO.